src/prefect/tasks/azure/datafactory.py

```diff
@@ -0,0 +1,356 @@
+        - location (str, optional): The location of the datafactory.
+        - options (dict, optional): The options to be passed to the
+            `create_or_update` method.
+        - **kwargs (dict, optional): Additional keyword arguments to pass to the
+            Task constructor.
+    """
+
+    def __init__(
+        self,
+        datafactory_name: str = None,
+        resource_group_name: str = None,
+        azure_credentials_secret: str = "AZ_CREDENTIALS",
+        location: str = "eastus",
+        options: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ) -> None:
+        self.datafactory_name = datafactory_name
+        self.resource_group_name = resource_group_name
+        self.azure_credentials_secret = azure_credentials_secret
+        self.location = location
+        self.options = options
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "datafactory_name",
+        "resource_group_name",
+        "azure_credentials_secret",
+        "location",
+        "options",
+    )
+    def run(
+        self,
+        datafactory_name: str = None,
+        resource_group_name: str = None,
+        azure_credentials_secret: str = None,
+        location: str = None,
+        options: Optional[Dict[str, Any]] = None,
+    ) -> Dict[Any, Any]:
+        """
+        Create an Azure datafactory.
+
+        Args:
+            - datafactory_name (str): Name of the datafactory to create.
+            - resource_group_name (str): Name of the resource group.
+            - azure_credentials_secret (str): Name of the Prefect Secret that stores
+                your Azure credentials; This Secret must be JSON string with the keys
+                `subscription_id`, `client_id`, `secret`, and `tenant`.
+            - location (str, optional): The location of the datafactory.
+            - options (dict, optional): The options to be passed to the
+                `create_or_update` method.
+
+        Returns:
+            The datafactory name.
+        """
+        if not datafactory_name:
+            raise ValueError("The datafactory_name must be specified.")
+        if not resource_group_name:
+            raise ValueError("The resource_group_name must be specified.")
+
+        datafactory_client = _get_datafactory_client(azure_credentials_secret)
+
+        self.logger.info(
+            f"Preparing to create the {datafactory_name} datafactory under "
+            f"{resource_group_name} in {location}"
+        )
+        factory = azure.mgmt.datafactory.models.Factory(location=location)
+        create_factory = datafactory_client.factories.create_or_update(
+            resource_group_name, datafactory_name, factory, **options or {}
+        )
+        while create_factory.provisioning_state != "Succeeded":
+            self.logger.info(
+                f"The {datafactory_name} factory status: "
+                f"{create_factory.provisioning_state}"
+            )
+            time.sleep(1)
```

Comment:
I think we could have a slightly higher poll interval to reduce the possibility of rate limiting. It'd also be nice if the poll interval was configurable.